pkg/iac/rego/scanner.go

```diff
@@ -17,12 +17,29 @@ import (
 
 	"github.com/aquasecurity/trivy/pkg/iac/debug"
 	"github.com/aquasecurity/trivy/pkg/iac/framework"
+	"github.com/aquasecurity/trivy/pkg/iac/providers"
 	"github.com/aquasecurity/trivy/pkg/iac/rego/schemas"
 	"github.com/aquasecurity/trivy/pkg/iac/scan"
 	"github.com/aquasecurity/trivy/pkg/iac/scanners/options"
 	"github.com/aquasecurity/trivy/pkg/iac/types"
 )
 
+var checkTypesWithSubtype = map[types.Source]struct{}{
+	types.SourceCloud:      {},
+	types.SourceDefsec:     {},
+	types.SourceKubernetes: {},
+}
+
+var supportedProviders map[string]struct{}
+
+func init() {
+	supportedProviders = make(map[string]struct{})
+	for _, p := range providers.AllProviders() {
+		supportedProviders[string(p)] = struct{}{}
+	}
+	supportedProviders["kind"] = struct{}{} // kubernetes
+}
```

Comment:
nit: Do we need this to be an `init` func? The abstraction into the `providers` package is nice but IMO not at the cost of keeping an `init` func.  I usually try to avoid having `init` funcs as they can get a bit problematic.

Reply:
fixed: https://github.com/aquasecurity/trivy/pull/6905/commits/9d2240aa82a41b5cfb19fe13ecd5a43a50b85663